test/lib/PlacesTestUtils.js

```diff
@@ -24,22 +26,69 @@ const PlacesTestUtils = Object.freeze({
    *            [optional] visitDate: visit date in microseconds from the epoch
    *            [optional] referrer: nsIURI of the referrer for this visit
    *          }
-   *
-   * @return {Promise}
-   *            resolves When all visits have been added successfully.
-   *            rejects JavaScript exception.
    */
   addVisits: Task.async(function*(placeInfo) {
-    let promise = new Promise((resolve, reject) => {
-      let places = [];
-      if (placeInfo instanceof Ci.nsIURI) {
-        places.push({uri: placeInfo});
-      } else if (Array.isArray(placeInfo)) {
-        places = places.concat(placeInfo);
-      } else {
-        places.push(placeInfo);
+    let places = [];
+
+    if (placeInfo instanceof Ci.nsIURI) {
+      places.push({uri: placeInfo});
+    } else if (Array.isArray(placeInfo)) {
+      places = places.concat(placeInfo);
+    } else {
+      places.push(placeInfo);
+    }
+
+    // Keep track of the number of rows for final validation
+    function* getPlacesRows() {
```

Comment:
This doesn't look right to me... if you want this to run async, you will need to also wrap all this in Task.spawn:

```JS
let initialNumRows = yield Task.spawn(function* getPlacesRows() {
  const conn = yield PlacesUtils.promiseDBConnection();
  const rows = yield conn.execute("SELECT id, url, title, frecency FROM moz_places");
  return rows;
}); 
```